cpp-ch/local-engine/Shuffle/PartitionWriter.cpp

```diff
@@ -300,5 +286,42 @@ void CelebornPartitionWriter::stop()
         shuffle_writer->split_result.total_bytes_written += item;
     }
 }
+
+void Partition::addBlock(DB::Block & block)
+{
+    std::unique_lock<std::mutex> lock(mtx, std::try_to_lock);
```

Comment:
更常用的写法 
std::lock_guard<std::mutex> guard{mtx}; 
/// do what you want without owns_lock judgement.

Reply:
if use std::lock_guard，it seems to lead to dead lock, as thread.join will suspend the main thread, if main thread hold the lock by lock_guard, but not finished, and the current thread try to get the lock which held by main thread, then the dead lock maybe appear. 

so i use try_to_lock, not to block if unable to get lock to avoid dead lock @taiyang-li

Reply:
This logic is not rgiht. If it doesn't get the lock, this block is not added in `blocks`

Reply:
OK, i will look into this, but the test result is correct